### PR TITLE
Search contract result details by eth hash

### LINKED ISSUE #3585: Add transaction hash path parameter to Get Contract Results REST API
### Problem

HIP-410 adds an `EthereumTransaction` with a 32 byte hash. We should allow searching for this hash via the REST API.

### Solution

* Update `/api/v1/contracts/results/{transactionId}` to accept a 64 character ethereum hash with optional `0x` prefix
* Query the new `ethereum_transaction` table by the hash to find its consensus timestamp

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by Ivo-Yankov**
**Description**:
Update the existing `/contracts/results/:transactionId` route to also accept an ethereum transaction hash with an optional `0x` prefix.

**Related issue(s)**:

Fixes #3585 

**Notes for reviewer**:

**Checklist**

- [X] Documented (Code comments, README, etc.)
- [X] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-rest/controllers/contractController.js [hunk 1]
```diff
@@ -892,18 +887,36 @@ class ContractController extends BaseController {
...
+    const {transactionIdOrHash} = req.params;
     const nonce = getLastNonceParamValue(req.query);
+    let transactions = [];
+    // When getting transactions, exclude duplicate transactions. there can be at most one
+    if (utils.isValidEthHash(transactionIdOrHash)) {
+      const ethHash = transactionIdOrHash.replace('0x', '');
+      // get transactions using ethereum hash and nonce
+      transactions = await TransactionService.getTransactionDetailsFromEthHash(
+        ethHash,
+        nonce,
```
Comment (xin-hedera) [inline hedera-mirror-rest/controllers/contractController.js:907]: `nonce` doesn't apply when searching transaction by hash

File: hedera-mirror-rest/controllers/contractController.js [hunk 2]
```diff
@@ -892,18 +887,36 @@ class ContractController extends BaseController {
...
+    if (utils.conflictingPathParam(req, 'transactionIdOrHash', 'logs')) {
+      return;
+    }
+
     utils.validateReq(req);
+
     // extract filters from query param
-    const transactionId = TransactionId.fromString(req.params.transactionId);
+    const {transactionIdOrHash} = req.params;
     const nonce = getLastNonceParamValue(req.query);
```
Comment (xin-hedera) [inline hedera-mirror-rest/controllers/contractController.js:899]: move this line to the `else` part where the path param is a transaction id.

File: hedera-mirror-rest/api/v1/openapi.yml [hunk 3]
```diff
@@ -3293,3 +3293,17 @@ components:
...
+      description: Transaction Id or a 32 byte hash with optional 0x prefix
+      examples: 
+        trahsactionId:
+          value: 0.0.10-1234567890-000000000
+        transactionHash:        
+          value: ddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef
+        transactionHashWith0xPrefix:        
+          value: '0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef'
+      schema:
+        type: string
```
Comment (xin-hedera) [inline hedera-mirror-rest/api/v1/openapi.yml:3309]: add a schema for transaction id string if there isn't one, then change this to
```suggestion
      schema:
        oneOf:
          - $ref: '#/components/schemas/EthereumHash'
          - $ref: '#/components/schemas/TransactionIdStr'
```

File: hedera-mirror-rest/service/transactionService.js [hunk 4]
```diff
@@ -38,14 +39,29 @@ class TransactionService extends BaseService {
...
+
+  static transactionDetailsFromTransactionIdQuery = `${this.selectTransactionDetailsBaseQuery}
     where ${Transaction.PAYER_ACCOUNT_ID} = $1
       and ${Transaction.VALID_START_NS} = $2`;
-  static transactionDetailsFromTransactionIdAndNonceQuery = `
-    ${TransactionService.transactionDetailsFromTransactionIdQuery}
-      and ${Transaction.NONCE} = $3`;
+
+  static transactionDetailsFromEthHashQuery = `with ${EthereumTransaction.tableAlias} as (
+      select ${EthereumTransaction.CONSENSUS_TIMESTAMP}, ${EthereumTransaction.HASH} from ${
```
Comment (xin-hedera) [inline hedera-mirror-rest/service/transactionService.js:54]:
```suggestion
      select ${EthereumTransaction.CONSENSUS_TIMESTAMP} from ${
```

File: hedera-mirror-rest/service/transactionService.js [hunk 5]
```diff
@@ -38,14 +39,29 @@ class TransactionService extends BaseService {
...
+    EthereumTransaction.tableName
+  }
+      where ${EthereumTransaction.tableName}.${EthereumTransaction.HASH} = $1
+  )
+    ${this.selectTransactionDetailsBaseQuery}
+    left join ${EthereumTransaction.tableAlias}
```
Comment (xin-hedera) [inline hedera-mirror-rest/service/transactionService.js:60]:
```suggestion
    join ${EthereumTransaction.tableAlias}
```

File: hedera-mirror-rest/service/transactionService.js [hunk 6]
```diff
@@ -38,14 +39,29 @@ class TransactionService extends BaseService {
...
+    on ${Transaction.getFullName(Transaction.CONSENSUS_TIMESTAMP)} = ${EthereumTransaction.getFullName(
+    EthereumTransaction.CONSENSUS_TIMESTAMP
+  )}
+    where ${EthereumTransaction.tableAlias}.${EthereumTransaction.HASH} = $1`;
```
Comment (xin-hedera) [inline hedera-mirror-rest/service/transactionService.js:64]:
```suggestion
  )}`;
```

File: hedera-mirror-rest/service/transactionService.js [hunk 7]
```diff
@@ -71,17 +87,31 @@ class TransactionService extends BaseService {
...
+    return this.getTransactionDetails(
+      TransactionService.transactionDetailsFromTransactionIdQuery,
+      [transactionId.getEntityId().getEncodedId(), transactionId.getValidStartNs()],
+      'getTransactionDetailsFromEthHash',
+      nonce,
+      excludeTransactionResults
+    );
+  }
+
+  async getTransactionDetailsFromEthHash(ethHash, nonce = undefined, excludeTransactionResults = []) {
```
Comment (xin-hedera) [inline hedera-mirror-rest/service/transactionService.js:100]: `nonce` doesn't apply

File: hedera-mirror-rest/api/v1/openapi.yml (hunk 3 above)
Comment (Nana-EC) [reply hedera-mirror-rest/api/v1/openapi.yml:3309]: One exists already `$ref: '#/components/schemas/TransactionId'`

Comment (xin-hedera) [reply hedera-mirror-rest/api/v1/openapi.yml:3309]: That's of type object not the string format we need here

File: hedera-mirror-rest/utils.js [hunk 8]
```diff
@@ -155,6 +155,16 @@ const isValidBlockHash = (query) => {
   return blockHashPattern.test(query);
 };
 
+const ethHashPattern = /^(0x)?([0-9A-Fa-f]{64})$/;
+const isValidEthHash = (hash) => {
```
Comment (Nana-EC) [inline hedera-mirror-rest/utils.js:159]: Add tests for this method

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Test request.
Other items already called out by other reviewers so will wait on that

**Review by xin-hedera — APPROVED** [review]
LGTM. just one nit

**Review by Nana-EC — APPROVED** [review]
LGTM

**Pull request merged**